### ISSUE TITLE
508-defect-3 [SCREENREADER, COGNITION]: State abbreviations SHOULD be coded as such

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#7808]
# [508-defect-3](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-3)

**Feedback framework**
- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

For all of the state abbreviations, they **should** be coded as abbreviations, providing the state name in the title. Sample text: If you live in AK, AZ, CA, CO, HI, ID, IA, KS, MN, MT, NB, NV, NM, ND, OK, OR, SD, TX, UT, WA, WY, etc… At present, it is confusing because it reads phonetically instead. It's likely that experienced screen reader users are used to this, but newly injured Veterans are likely to struggle as they are learning to use screen readers.

_**[2023-12-07 Fran] Note: Implementation of this was awaiting release of D10**_

## Point of Contact

**VFS Point of Contact:** Jennifer

## Acceptance Criteria

As a screen reader user, I want to clearly understand abbreviated text.

## Environment

* Operating System: all
* Browser: all
* Screenreading device: all
* Server destination: Devshop?/Prod?

## Steps to Recreate

1. Enter https://preview-prod.vfs.va.gov/preview?nodeId=3011 in browser (Va environment)
2. Start screenreading device listed in Environment
3. Navigate to the section with the abbreviated states
4. Verify that it is difficult to process the phonetically read abbreviations, especially in the long string of abbreviations.

## Possible Fixes (optional)

Recommend adding the following example for each abbreviation: 

```html
    <abbr title="Alaska">AK</abbr>, <abbr title="Arizona">Arizona</abbr>
111

## WCAG or Vendor Guidance (optional)

* [MDN <abbr>: The Abbreviation element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/abbr)
* [W3C Understanding WCAG 2.0 - Abbreviations](https://www.w3.org/TR/UNDERSTANDING-WCAG20/meaning-located.html)

**Comment by kelsonic** [comment]
I believe this is also a free text field in Drupal

![image](https://user-images.githubusercontent.com/12773166/82692687-632b7b00-9c1d-11ea-986c-ec67156e085d.png)

**Comment by jenstrickland** [comment]
This one is going to be a big effort. It may be helpful if I create a Github Markdown file with the code in it — or you could if you have time, so that the content folks will be able to copy and paste the states they need. It will also need a CSS style to remove the default `<abbr>` styling which I imagine design and PO will not love.

**Comment by kelsonic** [comment]
@RLHecht @DanielleThierryUSDSVA would either of you be able to implement the following on the page https://www.va.gov/decision-reviews/supplemental-claim/:

![image](https://user-images.githubusercontent.com/12773166/91736346-bdb46b80-eb6a-11ea-8747-fcab16443ffc.png)

```html
<abbr title="Alaska">AK</abbr>, <abbr title="Arizona">AZ</abbr>, <abbr title="California">CA</abbr>, <abbr title="Colorado">CO</abbr>, <abbr title="Hawaii">HI</abbr>, <abbr title="Idaho">ID</abbr>, <abbr title="Iowa">IA</abbr>, <abbr title="Kansas">KS</abbr>, <abbr title="Minnesota">MN</abbr>, <abbr title="Montana">MT</abbr>, <abbr title="Nebraska">NB</abbr>, <abbr title="Nevada">NV</abbr>, <abbr title="New Mexico">NM</abbr>, <abbr title="North Dakota">ND</abbr>, <abbr title="Oklahoma">OK</abbr>, <abbr title="Oregon">OR</abbr>, <abbr title="South Dakota">SD</abbr>, <abbr title="Texas">TX</abbr>, <abbr title="Utah">UT</abbr>, <abbr title="Washington">WA</abbr>, <abbr title="Wyoming">WY</abbr>
```

```html
<abbr title="Connecticut">CT</abbr>, <abbr title="Washington, D.C.">DC</abbr>, <abbr title="Delaware">DE</abbr>, <abbr title="Florida">FL</abbr>, <abbr title="Georgia">GA</abbr>, <abbr title="Massachusetts">MA</abbr>, <abbr title="Maryland">MD</abbr>, <abbr title="Maine">ME</abbr>, <abbr title="North Carolina">NC</abbr>, <abbr title="New Hampshire">NH</abbr>, <abbr title="New Jersey">NJ</abbr>, <abbr title="New York">NY</abbr>, <abbr title="Pennsylvania">PA</abbr>, <abbr title="Rhode Island">RI</abbr>, <abbr title="South Carolina">SC</abbr>, <abbr title="Virginia">VA</abbr>, <abbr title="Vermont">VT</abbr>, <abbr title="West Virginia">WV</abbr>
```

```html
<abbr title="Alabama">AL</abbr>, <abbr title="Arkansas">AR</abbr>, <abbr title="Illinois">IL</abbr>, <abbr title="Indiana">IN</abbr>, <abbr title="Kentucky">KY</abbr>, <abbr title="Louisiana">LA</abbr>, <abbr title="Michigan">MI</abbr>, <abbr title="Mississippi">MS</abbr>, <abbr title="Missouri">MO</abbr>, <abbr title="Ohio">OH</abbr>, <abbr title="Tennessee">TN</abbr>, or <abbr title="Wisconsin">WI</abbr>
```

**Comment by RLHecht** [comment]
Just tried entering in this is in the source field but it's not working. According to @kelsonic we can't make this update because the CMS is sanitizing the source. @MickinSahni should this be passed to the CMS team?

**Comment by jenstrickland** [comment]
@kevwalsh Do you know what we need to do to include the semantic markup in this field, if it is possible?

**Comment by DanielleThierryUSDSVA** [comment]
Hi @RLHecht! I happened to be searching for 508 tickets and came across this one. Just checking in to ask if there's anything we can help with on this?

**Comment by RLHecht** [comment]
@kelsonic is this something that you still need the CMS team to work on? From previous comments I think you and I tried to add this as source code and Drupal sanitized it like has been the case with other things we've tried. Do you know who should be taking the next steps on this?

**Comment by kelsonic** [comment]
Hey @RLHecht! 👋 So if we aren't able to add the following content in Drupal:

https://github.com/department-of-veterans-affairs/va.gov-team/issues/7808#issuecomment-683848451

then unfortunately I'm not sure how to achieve this. Perhaps we could modify how Drupal sanitizes the rich text so that it doesn't strip out these tags or properties, but that seems like a huge LOE + large risks compared to what we stand to gain from completing this ticket 😬

**Comment by joshkimux** [comment]
@laflannery for awareness!

**Comment by laflannery** [comment]
Will be able to be completed once D10 is launched

**Comment by jilladams** [comment]
@coforma-terry  @DanielleThierryUSDSVA @RLHecht FYI that the content updates requested here were previously blocked by Drupal 10 launch, which is now complete. Per Laura, CAIA should at this point be able to make these edits. I've added the sitewide-CAIA label to ticket, so y'all can prioritize / review. (Content may have drifted in this page since the ticket was created, not sure.)

If you do find you still hit technical blockers and need PW help, just let @FranECross and I know.

**Comment by RLHecht** [comment]
This is an old one! @jilladams it looks like the list of state abbreviations is no longer on this page so this ticket can be closed. Thanks for the follow-up!

**Comment by laflannery** [comment]
I love to close things! Thanks @RLHecht
